.travis.yml

```diff
@@ -58,16 +58,22 @@ before_install:
 - if [[ $TRAVIS_RUST_VERSION == "nightly" ]] && [[ $TRAVIS_OS_NAME == "windows" ]]; then rustup default nightly-msvc; fi
 
 before_script:
-  - rustup component add clippy
+  - if [[ $TRAVIS_RUST_VERSION != "nightly" ]]; then rustup component add clippy; fi
+
 
 script:
   - cargo test
   # TODO: enable GL backend
-  - (cd wgpu-core && cargo clippy --all-features)
-  - (cd wgpu-native && cargo clippy)
-  - (cd wgpu-remote && cargo clippy --all-features)
-  - if [[ $TRAVIS_OS_NAME == "osx" ]]; then (cd wgpu-native && cargo clippy --features vulkan-portability); fi
-  - if [[ $TRAVIS_OS_NAME == "linux" ]]; then cargo clippy --release; fi
+  - if [[ $TRAVIS_RUST_VERSION != "nightly" ]]; then (cd wgpu-core && cargo clippy --all-features); fi
```

Comment:
I think we should just have `BUILD_COMMAND` environment variable (equal to "clippy" or "check") that we set in the build matrix at the top of the file and just use here. Would you want to try that?